libs/content/workspace/src/tab/markdown_editor/bounds.rs

```diff
@@ -266,31 +263,35 @@ pub fn calc_links(buffer: &SubBuffer, text: &Text, ast: &Ast) -> PlainTextLinks
 }
 
 pub fn captured(
-    cursor: Cursor, paragraphs: &Paragraphs, ast: &Ast, text_range: &AstTextRange,
-    hover_syntax_reveal_debounce_state: HoverSyntaxRevealDebounceState, appearance: &Appearance,
-    cursor_paragraphs: (usize, usize),
+    cursor: Cursor, paragraphs: &Paragraphs, ast: &Ast, ast_text_range: &AstTextRange,
+    hover_syntax_reveal_debounce_state: &HoverSyntaxRevealDebounceState, appearance: &Appearance,
 ) -> bool {
-    let ast_node_range = ast.nodes[*text_range.ancestors.last().unwrap()].range;
-    let intersects_selection = ast_node_range.intersects_allow_empty(&cursor.selection);
-
-    let debounce_satisfied = hover_syntax_reveal_debounce_state.pointer_offset_updated_at
-        < Instant::now() - HOVER_SYNTAX_REVEAL_DEBOUNCE;
-    let intersects_pointer = debounce_satisfied
-        && hover_syntax_reveal_debounce_state
-            .pointer_offset
-            .map(|pointer_offset| {
-                ast_node_range.intersects(&(pointer_offset, pointer_offset), true)
-            })
-            .unwrap_or(false);
-
-    let text_range_paragraphs = paragraphs.find_intersecting(text_range.range, true);
-    let in_capture_disabled_paragraph = appearance.markdown_capture_disabled_for_cursor_paragraph
-        && cursor_paragraphs.intersects(&text_range_paragraphs, false);
-
-    match appearance.markdown_capture(text_range.node(ast).node_type()) {
+    if ast_text_range.range_type == AstTextRangeType::Text {
+        return false;
+    }
+
+    // check if this text range intersects any paragraph with selected text
+    let selection_paragraphs = paragraphs.find_intersecting(cursor.selection, true);
+    let text_range_paragraphs = paragraphs.find_intersecting(ast_text_range.range, true);
+    let intersects_selected_paragraph =
+        selection_paragraphs.intersects(&text_range_paragraphs, false);
+
+    // check if the pointer is hovering over the ast leaf node of this text range
+    let now = hover_syntax_reveal_debounce_state.frame_start;
+    let time_to_hover_reveal_syntax =
+        hover_syntax_reveal_debounce_state.updated_at + HOVER_SYNTAX_REVEAL_DEBOUNCE;
+    let debounce_satisfied = now > time_to_hover_reveal_syntax;
+    let intersects_pointer = hover_syntax_reveal_debounce_state
+        .pointer_ast_leaf_node
+        .map(|pointer_ast_leaf_node| {
+            *ast_text_range.ancestors.last().unwrap() == pointer_ast_leaf_node
+        })
+        .unwrap_or_default();
+
+    match appearance.markdown_capture(ast_text_range.node(ast).node_type()) {
         CaptureCondition::Always => true,
         CaptureCondition::NoCursor => {
-            !(intersects_selection || intersects_pointer || in_capture_disabled_paragraph)
+            !((debounce_satisfied && intersects_pointer) || intersects_selected_paragraph)
```

Comment:
`intersects_selection` was deleted because `in_capture_disabled_paragraph` (now called `intersects_selected_paragraph`) is always true if `intersects_selection` is true